internal/discover/hooks.go

```diff
@@ -44,26 +44,66 @@ func (h *Hook) Hooks() ([]Hook, error) {
 	return []Hook{*h}, nil
 }
 
+type HookName string
+
+// DisabledHooks allows individual hooks to be disabled.
+type DisabledHooks map[HookName]bool
+
+func (d DisabledHooks) Set(value HookName) {
+	if value == "all" {
+		for _, hook := range AllHooks {
+			d[hook] = true
+		}
+		return
+	}
+
+	d[value] = true
+}
+
+const (
+	// HookEnableCudaCompat refers to the hook used to enable CUDA Forward Compatibility.
+	// This was added with v1.17.5 of the NVIDIA Container Toolkit.
+	HookEnableCudaCompat = HookName("enable-cuda-compat")
+	// directory path to be mounted into a container.
+	HookCreateSymlinks = HookName("create-symlinks")
+	// HookUpdateLDCache refers to the hook used to  Update the dynamic linker
+	// cache inside the directory path to be mounted into a container.
+	HookUpdateLDCache = HookName("update-ldcache")
+)
+
+// AllHooks maintains a future-proof list of all defined hooks.
+var AllHooks = []HookName{
+	HookEnableCudaCompat,
+	HookCreateSymlinks,
+	HookUpdateLDCache,
+}
+
 // Option is a function that configures the nvcdilib
 type Option func(*CDIHook)
 
 type CDIHook struct {
 	nvidiaCDIHookPath string
+	disabledHooks     DisabledHooks
 }
 
 type HookCreator interface {
-	Create(string, ...string) *Hook
+	Create(HookName, ...string) *Hook
 }
 
-func NewHookCreator(nvidiaCDIHookPath string) HookCreator {
+func NewHookCreator(nvidiaCDIHookPath string, disabledHooks DisabledHooks) HookCreator {
```

Comment:
What about using:
```
```suggestion
func NewHookCreator[T string|HookName](nvidiaCDIHookPath string, disabledHooks ...T) HookCreator {
```

The fact that we're using a map should be internal to this pacakge and not exposed through the interface.

Reply:
Looks good. I have added a commit on top that also sets the hook path as an option for consistency.